rfcs/0185-redistribute-redistributable.md

```diff
@@ -0,0 +1,108 @@
+---
+feature: redistribute-redistributable
+start-date: 2024-12-15
+author: Ekleog
+co-authors: (find a buddy later to help out with the RFC)
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: https://github.com/NixOS/nixpkgs/issues/83884
+---
+
+# Summary
+[summary]: #summary
+
+Make Hydra build and provide all redistributable software.
+
+# Motivation
+[motivation]: #motivation
+
+Currently, Hydra builds only free software and unfree redistributable firmware.
+This means that unfree redistributable software needs to be rebuilt by all the users.
+For example, using MongoDB on a Raspberry Pi 4 (aarch64, which otherwise has access to hydra's cache) takes literally days and huge amounts of swap.
+
+Hydra could provide builds for unfree redistributable software, at minimal added costs.
+This would make life much better for users of such software.
+Especially when the software is still source-available even without being free software, like MongoDB.
+
+# Detailed design
+[design]: #detailed-design
+
+Hydra will build all packages with licenses for which `redistributable = true`.
+
+# Examples and Interactions
+[examples-and-interactions]: #examples-and-interactions
+
+With this change, Hydra will start building, among others:
+- CUDA
+- DragonflyDB
+- MongoDB
+- Nomad
+- NVIDIA drivers
+- Outline
+- SurrealDB
+- TeamSpeak
+- Terraform
+- Unrar
+- Vagrant
+
+Hydra will keep not building, among others:
+- CompCert
+- DataBricks
+- Elasticsearch
+- GeoGebra
+- Widevine CDM
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+The only previously listed drawback is that NixOS could end up including unfree software in the ISO image without noticing.
+However, as there is already unfree firmware, this fight is already half-lost.
+
+Also, adding unfree software to the ISO image would still require a NixOS maintainer to actually add it there.
+The only benefit we currently get out of not building unfree redistributable software, is that the hydra builds for the ISO would fail if someone were to make a mistake.
+
+# Alternatives
+[alternatives]: #alternatives
+
+### Having Hydra actually only build FOSS derivations, not even unfree redistributable firmware
+
+This would likely break many installation scenarios, but would bring us to a consistent ethical standpoint, though it's not mine.
+
+### Keeping the statu quo
+
+This results in very long builds for lots of software, as exhibited by the number of years people have been complaining about it.
+
+### Implementing this RFC
+
+See above for the details
```

Comment:
By definition, "implementing this RFC" is not an alternative to this RFC. Besides, the "see above" reference doesn't provide additional information.

```suggestion
```

Reply:
I hesitated, then added it here to make it obvious that this is a scale ranging from "allow nothing unfree" to "allow everything unfree", with various marks on the scale. Was it not clear enough? I can try rewording the whole section if yes.

Reply:
It might make this section more readable to put a summary of what the RFC is in the title and say "This is what this RFC suggests" or similar in the description.

Referencing "this RFC" in the titles here does feel odd especially since this RFC might change (e.g. By specifying a way to ensure Hydra doesn't build an ISO with unfree software).